dwave/optimization/src/nodes/manipulation.cpp

```diff
@@ -578,6 +578,107 @@ std::pair<double, double> ReshapeNode::minmax(
 
 void ReshapeNode::revert(State& state) const {}  // stateless node
 
+ResizeNode::ResizeNode(ArrayNode* array_ptr, std::vector<ssize_t>&& shape, double fill_value)
+        : ArrayOutputMixin(shape), array_ptr_(array_ptr), fill_value_(fill_value) {
+    // our incoming array can be any shape/size, but we cannot be dynamic
+    if (this->dynamic()) throw std::invalid_argument("cannot resize to a dynamic shape");
+
+    // determine whether our fill value will ever matter. We need this info later for
+    // min/max/integral. We could do this lazily, but let's just be proactive for
+    // simplicity
+    if (array_ptr_->dynamic()) {
+        // We'll never use the fill value of the minimize size of our predecessor
+        // array is greater than or equal to our size.
+        const ssize_t min_size = array_ptr_->sizeinfo().substitute(100).min.value_or(0);
```

Comment:
also magic number for max_depth here..

Reply:
yes, the same magic number used everywhere else. There is a bigger fix forthcoming: https://github.com/dwavesystems/dwave-optimization/issues/353